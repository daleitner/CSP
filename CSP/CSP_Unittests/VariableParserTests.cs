using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ApprovalTests;
using ApprovalTests.Reporters;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using CSP;

namespace CSP_Unittests
{
	[TestClass]
	[UseReporter(typeof(WinMergeReporter))]
	public class VariableParserTests
	{
		[TestMethod]
		public void VerifyStringWithOneVariable()
		{
			var input = "a";
			var output = VariableParser.Parse(input);
			Approvals.VerifyAll(output, "Variable");
		}

		[TestMethod]
		public void VerifyStringWithTwoVariables()
		{
			var input = "a,b";
			var output = VariableParser.Parse(input);
			Approvals.VerifyAll(output, "Variable");
		}

		[TestMethod]
		public void VerifyStringWithUntrimmedVariables()
		{
			var input = " a,b ";
			var output = VariableParser.Parse(input);
			Approvals.VerifyAll(output, "Variable");
		}

		[TestMethod]
		public void WhenVariableHasWhiteSpace_ThenReplaceWithUnderline()
		{
			var input = " a,b c";
			var output = VariableParser.Parse(input);
			Approvals.VerifyAll(output, "Variable");
		}

		[TestMethod]
		public void WhenVariablesNotUnique_ThenThrowException()
		{
			var input = " a,a";
			var exc = "";
			try
			{
				var output = VariableParser.Parse(input);
			}
			catch (ArgumentException ex)
			{
				exc = ex.Message;
			}
			Approvals.Verify("Exception message: " +exc);
		}

		[TestMethod]
		public void WhenVariableLengthIsNull_ThenThrowException()
		{
			var input = " a,";
			var exc = "";
			try
			{
				var output = VariableParser.Parse(input);
			}
			catch (ArgumentException ex)
			{
				exc = ex.Message;
			}
			Approvals.Verify("Exception message: " + exc);
		}

		[TestMethod]
		public void WhenInputIsEmpty_ThenThrowException()
		{
			var input = "";
			var exc = "";
			try
			{
				var output = VariableParser.Parse(input);
			}
			catch (ArgumentException ex)
			{
				exc = ex.Message;
			}
			Approvals.Verify("Exception message: " + exc);
		}
	}
}
