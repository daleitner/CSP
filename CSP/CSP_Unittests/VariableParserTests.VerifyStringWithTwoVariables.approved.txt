Variable[0] = a
Variable[1] = b
